golden-tests/old-derivation/flake.nix

```diff
@@ -0,0 +1,22 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
```

Comment:
Let's pin a specific revision here to avoid unexpected changes of `flake.lock`

Reply:
It already pinned in flake.lock. You can change pin by editing flake.lock or doing `nix flake update`/`nix flake lock --update-input nixpkgs`, all of these ways will cause changing of flake.lock file. 
nixpkgs input in flake will be sourced from outside only if it will not explicitly listed in inputs.

Reply:
Yes, one can accidentally run `nix flake lock --update-input nixpkgs` in one of derivations, but not in the other, and then tests will likely start to fail

Reply:
He will be wrong, but this change will be shown in changes of git, so I think, that this will not cause any critical errors. And what will stop him from changing one of fleake.nix?

Reply:
> And what will stop him from changing one of fleake.nix?

You can write something like `# DON'T CHANGE THE PINNED REVISION BELOW WITHOUT REGENERATING THE 'expected-output'` and then it's much less likely be changed on accident :sweat_smile:

Reply:
My point is that if some dependency implicit change can be breaking, it should be pinned explicitly

Reply:
But it isn't implicit, because flake.lock is in the same git repo.

Reply:
Okay, I don't insist

Reply:
Let's get your scenario. We wanted to add tests to the CI. Someone will
1) type command `cd ./golden-tests/new-derivation; nix flake lock --update-input nixpkgs`
2) make a PR
3) CI would report about an error.
4) In changes of commit would be clearly shown changes of one flake.lock.